src/main/java/fr/insalyon/creatis/moteurlite/MoteurLite.java

```diff
@@ -0,0 +1,174 @@
+package fr.insalyon.creatis.moteurlite;
+
+import java.io.File;
+import java.net.URI;
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+
+import com.fasterxml.jackson.databind.ObjectMapper;
+
+import fr.insalyon.creatis.gasw.Gasw;
+import fr.insalyon.creatis.gasw.GaswInput;
+import fr.insalyon.creatis.moteurlite.boutiques.BoutiquesDescriptor;
+import fr.insalyon.creatis.moteurlite.boutiques.Input;
+import fr.insalyon.creatis.moteurlite.inputsParser.InputsFileParser;
+import fr.insalyon.creatis.moteurlite.iterationStrategy.IterationStrategy;
+
+
+/**
+ * 
+ * Author: Sandesh Patil [https://github.com/sandepat]
+ * 
+ */
+
+public class MoteurLite {
+    private Gasw gasw = Gasw.getInstance();
+    private IterationStrategy iterationStrategy = new IterationStrategy();
+    private Workflowsdb workflowsdb = new Workflowsdb();
+
+    private int sizeOfInputs;
+    private String applicationName;
+    private String boutiquesFilePath;
+    private String executableName;
+    private HashMap<Integer, String> inputBoutiquesId;
+    private String workflowId;
+    String inputsFilePath;
+
+    private Map<String, String> inputTypes = new HashMap<>();
+    private Map<String, String> resultsDirectory;
+    private List<Map<String, String>> inputMap;
+
+    private List<Map<String, String>> inputData;
+    
+
+
+    public static void main(String[] args) throws Exception {
+        new MoteurLite(args);
+    }
+
+    public MoteurLite(String[] args) throws Exception {
+        // Verify arguments
+        if (args.length != 3) {
+            throw new IllegalArgumentException("Exactly 3 arguments are required: workflowId, boutiquesFilePath, inputsFilePath.");
+        }
+        workflowId = args[0];
+        boutiquesFilePath = args[1];
+        inputsFilePath = args[2];
+
+        checkArguments(args);
+
```

Comment:
The rest of the main method should look like this (small changes may be needed of course) :
This will indicate how to refactor stuff.
No class field should be needed anymore.

```
// build dependencies
WorkflowsDbRepository workflowsDbRepo = new WorkflowsDbRepository();
BoutiquesService boutiquesService = new BoutiquesService();
IterationStrategyService iterationStrategyService = new IterationStrategyService(boutiquesService);

// parse boutiques and inputs
BoutiquesDescriptor descriptor = boutiquesService.parseFile(boutiquesFilePath);
Map<String,List<String>> inputValues = new InputsFileParser().parse(inputsFilePath);

// init db
workflowsDbRepo.persistInputs(workflowId, inputValues, boutiqueService.getInputTypes(descriptor));

// compute iteration strategy
List<Map<String,String>> jobsInputValues = iterationStrategyService.compute(descriptor, inputValues);

// init gasw
Gasw gasw = Gasw.getInstance();
GaswMonitor gaswMonitor = new GaswMonitor(workflowId, descriptor.getName(), jobsInputValues.size(), gasw);
gasw.setNotificationClient(gaswMonitor);
gaswMonitor.start();

// launch jobs
createJobs(descriptor, jobsInputValues);

```